2022-10-25 12:28:06 info: ---------------------------------------
2022-10-25 12:28:06 info: >>> Running Unit Test `ex1-johnWho` <<<
2022-10-25 12:28:06 info: ---------------------------------------
2022-10-25 12:28:08 error: *** Unit Test Error Report ***

Command failed: npx jest ex1-johnWho.test.js --colors
 FAIL  3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js
  getAnonName
    ✓ should exist and be executable (2 ms)
    ✕ should have all TODO comments removed (1 ms)
    ✓ `getAnonName` should not contain unneeded console.log calls (1 ms)
    ✓ should call `new Promise()`
    ✓ should take a single argument (1 ms)
    ✓ `resolve()` should be called with a one argument
    ✓ `reject()` should be called with a one argument (1 ms)
    ✓ should resolve when called with a string argument (1 ms)
    ✓ should reject with an Error object when called without an argument

  ● getAnonName › should have all TODO comments removed

    expect(received).toBeFalsy()

    Received: true

      103 | function testTodosRemoved(getSource) {
      104 |   test('should have all TODO comments removed', () => {
    > 105 |     expect(/\bTODO\b/.test(getSource())).toBeFalsy();
          |                                          ^
      106 |   });
      107 | }
      108 |

      at Object.toBeFalsy (test-runner/unit-test-helpers.js:105:42)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 8 passed, 9 total
Snapshots:   0 total
Time:        0.63 s
Ran all test suites matching /ex1-johnWho.test.js/i.
2022-10-25 12:28:10 error: *** ESLint Report ***

ex1-johnWho.js
  23:1  warning  Remove commented-out code  hyf/no-commented-out-code
  30:5  warning  Remove commented-out code  hyf/no-commented-out-code
  32:5  warning  Remove commented-out code  hyf/no-commented-out-code

✖ 3 problems (0 errors, 3 warnings)


2022-10-25 12:29:35 info: ---------------------------------------
2022-10-25 12:29:35 info: >>> Running Unit Test `ex1-johnWho` <<<
2022-10-25 12:29:35 info: ---------------------------------------
2022-10-25 12:29:36 error: *** Unit Test Error Report ***

Command failed: npx jest ex1-johnWho.test.js --colors
 FAIL  3-UsingAPIs/Week1/unit-tests/ex1-johnWho.test.js
  getAnonName
    ✓ should exist and be executable (2 ms)
    ✕ should have all TODO comments removed (1 ms)
    ✓ `getAnonName` should not contain unneeded console.log calls (7 ms)
    ✓ should call `new Promise()`
    ✓ should take a single argument (1 ms)
    ✓ `resolve()` should be called with a one argument
    ✓ `reject()` should be called with a one argument
    ✓ should resolve when called with a string argument (1 ms)
    ✓ should reject with an Error object when called without an argument (1 ms)

  ● getAnonName › should have all TODO comments removed

    expect(received).toBeFalsy()

    Received: true

      103 | function testTodosRemoved(getSource) {
      104 |   test('should have all TODO comments removed', () => {
    > 105 |     expect(/\bTODO\b/.test(getSource())).toBeFalsy();
          |                                          ^
      106 |   });
      107 | }
      108 |

      at Object.toBeFalsy (test-runner/unit-test-helpers.js:105:42)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 8 passed, 9 total
Snapshots:   0 total
Time:        0.369 s, estimated 1 s
Ran all test suites matching /ex1-johnWho.test.js/i.
2022-10-25 12:29:37 error: *** ESLint Report ***

ex1-johnWho.js
  23:1  warning  Remove commented-out code  hyf/no-commented-out-code
  30:5  warning  Remove commented-out code  hyf/no-commented-out-code
  32:5  warning  Remove commented-out code  hyf/no-commented-out-code

✖ 3 problems (0 errors, 3 warnings)


2022-10-25 12:30:01 info: ---------------------------------------
2022-10-25 12:30:01 info: >>> Running Unit Test `ex1-johnWho` <<<
2022-10-25 12:30:01 info: ---------------------------------------
2022-10-25 12:30:02 info: All unit tests passed.
2022-10-25 12:30:05 info: All steps were completed successfully
2022-10-25 12:50:23 info: -------------------------------------------------
2022-10-25 12:50:23 info: >>> Running Unit Test `ex2-checkDoubleDigits` <<<
2022-10-25 12:50:23 info: -------------------------------------------------
2022-10-25 12:50:24 error: *** Unit Test Error Report ***

Command failed: npx jest ex2-checkDoubleDigits.test.js --colors
 FAIL  3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js
  checkDoubleDigits
    ✓ should exist and be executable (2 ms)
    ✓ should have all TODO comments removed
    ✓ `checkDoubleDigits` should not contain unneeded console.log calls (1 ms)
    ✓ should call new Promise() (1 ms)
    ✓ `resolve()` should be called with a one argument
    ✓ `reject()` should be called with a one argument (1 ms)
    ✓ should be a function that takes a single argument
    ✕ (9) should return a rejected promise with an Error object (2 ms)
    ✕ (10) should return a promise that resolves to "This is a double digit number!" (1 ms)
    ✕ (99) should return a promise that resolves to "This is a double digit number!" (1 ms)
    ✕ (100) should return a rejected promise with an Error object (1 ms)

  ● checkDoubleDigits › (9) should return a rejected promise with an Error object

    expect(received).rejects.toBeInstanceOf(expected)

    Expected constructor: Error

    Received value has no prototype
    Received value: "Expected a double digit number but got 9"

      65 |     const promise = checkDoubleDigits(9);
      66 |     expect(promise).toBeInstanceOf(Promise);
    > 67 |     return expect(promise).rejects.toBeInstanceOf(Error);
         |                                    ^
      68 |   });
      69 |
      70 |   test('(10) should return a promise that resolves to "This is a double digit number!"', () => {

      at Object.toBeInstanceOf (node_modules/expect/build/index.js:241:22)
      at Object.toBeInstanceOf (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:67:36)

  ● checkDoubleDigits › (10) should return a promise that resolves to "This is a double digit number!"

    expect(received).resolves.toEqual()

    Received promise rejected instead of resolved
    Rejected to value: "Expected a double digit number but got 10"

      73 |     const promise = checkDoubleDigits(10);
      74 |     expect(promise).toBeInstanceOf(Promise);
    > 75 |     return expect(promise).resolves.toEqual(
         |            ^
      76 |       expect.stringContaining('This is a double digit number!')
      77 |     );
      78 |   });

      at expect (node_modules/expect/build/index.js:128:15)
      at Object.expect (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:75:12)

  ● checkDoubleDigits › (10) should return a promise that resolves to "This is a double digit number!"

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      69 |
      70 |   test('(10) should return a promise that resolves to "This is a double digit number!"', () => {
    > 71 |     expect.assertions(3);
         |            ^
      72 |     expect(exported).toBeDefined();
      73 |     const promise = checkDoubleDigits(10);
      74 |     expect(promise).toBeInstanceOf(Promise);

      at Object.assertions (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:71:12)

  ● checkDoubleDigits › (99) should return a promise that resolves to "This is a double digit number!"

    expect(received).resolves.toEqual()

    Received promise rejected instead of resolved
    Rejected to value: "Expected a double digit number but got 99"

      82 |     const promise = checkDoubleDigits(99);
      83 |     expect(promise).toBeInstanceOf(Promise);
    > 84 |     return expect(promise).resolves.toEqual(
         |            ^
      85 |       expect.stringContaining('This is a double digit number!')
      86 |     );
      87 |   });

      at expect (node_modules/expect/build/index.js:128:15)
      at Object.expect (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:84:12)

  ● checkDoubleDigits › (99) should return a promise that resolves to "This is a double digit number!"

    expect.assertions(2)

    Expected two assertions to be called but received one assertion call.

      79 |
      80 |   test('(99) should return a promise that resolves to "This is a double digit number!"', () => {
    > 81 |     expect.assertions(2);
         |            ^
      82 |     const promise = checkDoubleDigits(99);
      83 |     expect(promise).toBeInstanceOf(Promise);
      84 |     return expect(promise).resolves.toEqual(

      at Object.assertions (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:81:12)

  ● checkDoubleDigits › (100) should return a rejected promise with an Error object

    expect(received).rejects.toBeInstanceOf(expected)

    Expected constructor: Error

    Received value has no prototype
    Received value: "Expected a double digit number but got 100"

      91 |     const promise = checkDoubleDigits(100);
      92 |     expect(promise).toBeInstanceOf(Promise);
    > 93 |     return expect(promise).rejects.toBeInstanceOf(Error);
         |                                    ^
      94 |   });
      95 | });
      96 |

      at Object.toBeInstanceOf (node_modules/expect/build/index.js:241:22)
      at Object.toBeInstanceOf (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:93:36)

Test Suites: 1 failed, 1 total
Tests:       4 failed, 7 passed, 11 total
Snapshots:   0 total
Time:        0.509 s
Ran all test suites matching /ex2-checkDoubleDigits.test.js/i.
2022-10-25 12:52:56 info: -------------------------------------------------
2022-10-25 12:52:56 info: >>> Running Unit Test `ex2-checkDoubleDigits` <<<
2022-10-25 12:52:56 info: -------------------------------------------------
2022-10-25 12:52:58 error: *** Unit Test Error Report ***

Command failed: npx jest ex2-checkDoubleDigits.test.js --colors
 FAIL  3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js
  checkDoubleDigits
    ✓ should exist and be executable (1 ms)
    ✓ should have all TODO comments removed (1 ms)
    ✓ `checkDoubleDigits` should not contain unneeded console.log calls (1 ms)
    ✓ should call new Promise()
    ✓ `resolve()` should be called with a one argument
    ✓ `reject()` should be called with a one argument
    ✓ should be a function that takes a single argument
    ✕ (9) should return a rejected promise with an Error object (2 ms)
    ✓ (10) should return a promise that resolves to "This is a double digit number!" (1 ms)
    ✓ (99) should return a promise that resolves to "This is a double digit number!" (1 ms)
    ✕ (100) should return a rejected promise with an Error object (1 ms)

  ● checkDoubleDigits › (9) should return a rejected promise with an Error object

    expect(received).rejects.toBeInstanceOf(expected)

    Expected constructor: Error

    Received value has no prototype
    Received value: "Expected a double digit number but got 9"

      65 |     const promise = checkDoubleDigits(9);
      66 |     expect(promise).toBeInstanceOf(Promise);
    > 67 |     return expect(promise).rejects.toBeInstanceOf(Error);
         |                                    ^
      68 |   });
      69 |
      70 |   test('(10) should return a promise that resolves to "This is a double digit number!"', () => {

      at Object.toBeInstanceOf (node_modules/expect/build/index.js:241:22)
      at Object.toBeInstanceOf (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:67:36)

  ● checkDoubleDigits › (100) should return a rejected promise with an Error object

    expect(received).rejects.toBeInstanceOf(expected)

    Expected constructor: Error

    Received value has no prototype
    Received value: "Expected a double digit number but got 100"

      91 |     const promise = checkDoubleDigits(100);
      92 |     expect(promise).toBeInstanceOf(Promise);
    > 93 |     return expect(promise).rejects.toBeInstanceOf(Error);
         |                                    ^
      94 |   });
      95 | });
      96 |

      at Object.toBeInstanceOf (node_modules/expect/build/index.js:241:22)
      at Object.toBeInstanceOf (3-UsingAPIs/Week1/unit-tests/ex2-checkDoubleDigits.test.js:93:36)

Test Suites: 1 failed, 1 total
Tests:       2 failed, 9 passed, 11 total
Snapshots:   0 total
Time:        0.473 s, estimated 1 s
Ran all test suites matching /ex2-checkDoubleDigits.test.js/i.
2022-10-25 01:06:03 info: -------------------------------------------------
2022-10-25 01:06:03 info: >>> Running Unit Test `ex2-checkDoubleDigits` <<<
2022-10-25 01:06:03 info: -------------------------------------------------
2022-10-25 01:06:05 info: All unit tests passed.
2022-10-25 01:06:07 info: All steps were completed successfully
2022-10-25 01:28:18 info: ---------------------------------------
2022-10-25 01:28:18 info: >>> Running Unit Test `ex3-rollDie` <<<
2022-10-25 01:28:18 info: ---------------------------------------
2022-10-25 01:28:20 error: *** Unit Test Error Report ***

Command failed: npx jest ex3-rollDie.test.js --colors
 FAIL  3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js
  rollDie
    ✓ should exist and be executable (2 ms)
    ✕ should have all TODO comments removed (1 ms)
    ✓ should call `new Promise()`
    ✓ `resolve()` should be called with a one argument
    ✓ `reject()` should be called with a one argument (1 ms)
    ✓ should resolve when the die settles successfully (1 ms)
    ✓ should reject with an Error when the die rolls off the table (1 ms)

  ● rollDie › should have all TODO comments removed

    expect(received).toBeFalsy()

    Received: true

      103 | function testTodosRemoved(getSource) {
      104 |   test('should have all TODO comments removed', () => {
    > 105 |     expect(/\bTODO\b/.test(getSource())).toBeFalsy();
          |                                          ^
      106 |   });
      107 | }
      108 |

      at Object.toBeFalsy (test-runner/unit-test-helpers.js:105:42)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 6 passed, 7 total
Snapshots:   0 total
Time:        0.485 s
Ran all test suites matching /ex3-rollDie.test.js/i.
2022-10-25 01:29:25 info: ---------------------------------------
2022-10-25 01:29:25 info: >>> Running Unit Test `ex3-rollDie` <<<
2022-10-25 01:29:25 info: ---------------------------------------
2022-10-25 01:29:27 error: *** Unit Test Error Report ***

Command failed: npx jest ex3-rollDie.test.js --colors
 FAIL  3-UsingAPIs/Week1/unit-tests/ex3-rollDie.test.js
  rollDie
    ✓ should exist and be executable (3 ms)
    ✕ should have all TODO comments removed (2 ms)
    ✓ should call `new Promise()`
    ✓ `resolve()` should be called with a one argument
    ✓ `reject()` should be called with a one argument (1 ms)
    ✓ should resolve when the die settles successfully (1 ms)
    ✓ should reject with an Error when the die rolls off the table (1 ms)

  ● rollDie › should have all TODO comments removed

    expect(received).toBeFalsy()

    Received: true

      103 | function testTodosRemoved(getSource) {
      104 |   test('should have all TODO comments removed', () => {
    > 105 |     expect(/\bTODO\b/.test(getSource())).toBeFalsy();
          |                                          ^
      106 |   });
      107 | }
      108 |

      at Object.toBeFalsy (test-runner/unit-test-helpers.js:105:42)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 6 passed, 7 total
Snapshots:   0 total
Time:        0.572 s, estimated 1 s
Ran all test suites matching /ex3-rollDie.test.js/i.
2022-10-25 01:32:12 info: ---------------------------------------
2022-10-25 01:32:12 info: >>> Running Unit Test `ex3-rollDie` <<<
2022-10-25 01:32:12 info: ---------------------------------------
2022-10-25 01:32:14 info: All unit tests passed.
2022-10-25 01:32:17 info: All steps were completed successfully
